chalk-solve/src/infer/unify.rs

```diff
@@ -111,6 +111,7 @@ impl<'t, I: Interner> Unifier<'t, I> {
         }
 
         match (a.kind(interner), b.kind(interner)) {
+            (TyKind::Error, _) | (_, TyKind::Error) => Ok(()),
```

Comment:
This is a *bit* inconsistent with the relation in rustc. Could you move this below the infer var relate branches? 

That is to say, when we relate `?0 eq Error`, in rustc we set `?0 := Error`. But in this case, it leaves the infer var unresolved.

Reply:
ah well no, let me move it further down